src/htcondorce/registry.py

```diff
@@ -77,7 +77,14 @@ def osgid_to_ce(osgid):
 def fetch_tokens(reqid, config):
     binary = config.get('CONDOR_TOKEN_REQUEST_LIST', 'condor_token_request_list')
     pool = config.get('CONDORCE_COLLECTOR')
-    args = [binary, '-reqid', str(reqid), '-json']
+
+    try:
+        assert int(reqid) > 0
+    except (ValueError, AssertionError):
+        raise CondorToolException("Received invalid code: %s" % reqid)
+    else:
+        args = [binary, '-reqid', str(reqid), '-json']
+
```

Comment:
What is the explicit str conversion for?  (`str(reqid)`)  Is `reqid` sometimes input as an `int` object?  (I see it's always `request.form.get('code')`, which i would have guessed was already a `str`)

Anyway looks fine how you have it.  Might be tempting to tuck the try/except check into its own function, but that's up to you.

Reply:
> What is the explicit str conversion for? (str(reqid)) Is reqid sometimes input as an int object? (I see it's always request.form.get('code'), which i would have guessed was already a str)

Yeah, I imagine you're correct but I don't think I want to change it for the time being. In the Python 3 version I'll add some type annotations

> Anyway looks fine how you have it. Might be tempting to tuck the try/except check into its own function, but that's up to you.

Done!